### PR TITLE
Include sort parameter in beatmapset search response

### PR DESCRIPTION
**Pull request by nanaya**
With advanced search, non-blank `query`/`q` parameter doesn't always default to `relevance` anymore. Duplicating the parsing logic to the frontend doesn't seem to make sense either so it's probably best to render current sort option based on server response.

Well, I tried updating the frontend part to do it but it's a bit more complicated. So here's just the response for starter.

### CODE REVIEW

**Comment by peppy** [comment]
Seems like something we want to get merged now that advanced search is already live?

**Comment by nanaya** [comment]
The frontend also need to be hooked up. And then there's semi-related bug recently found where sort option is reset when search query is updated.

File: app/Libraries/Search/BeatmapsetSearchRequestParams.php [hunk 1]
```diff
@@ -274,21 +283,23 @@ private function parseQuery(): void
...
 
-        if (empty($field)) {
-            $this->sorts = $this->getDefaultSort($order);
-        } else {
-            $this->sorts = $this->normalizeSort(new Sort($field, $order));
+        if (!in_array($this->sortOrder, ['asc', 'desc'], true)) {
+            $this->sortOrder = 'desc';
         }
 
+        $this->sorts = $this->normalizeSort(new Sort($mappedField, $this->sortOrder));
```
Comment (notbakaneko) [inline app/Libraries/Search/BeatmapsetSearchRequestParams.php:299]: maybe make `remapSortField` non-static (and rename?) and assign `$this->sortField` there? The way it bounces between `$mappedField` and `$this->sortField` (and then `sortOrder` in the middle) reads a bit confusing.

**Review by notbakaneko — CHANGES_REQUESTED** [review]

File: app/Libraries/Search/BeatmapsetSearchRequestParams.php (hunk 1 above)
Comment (nanaya) [reply app/Libraries/Search/BeatmapsetSearchRequestParams.php]: doing that in the `normalizeSort` (now `setSorts`) seems like a better idea :man_dancing:

**Review by notbakaneko — APPROVED** [review]

**Pull request merged**